stubs/CoreImmutableClasses.phpstub

```diff
@@ -1,11 +1,111 @@
 <?php
 
-/**
- * @psalm-immutable
- */
 class DateTimeImmutable implements DateTimeInterface
 {
     public function __construct(string $datetime = "now", DateTimeZone $timezone = null) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static|false
+     */
+    public static function createFromFormat(string $format, string $datetime, ?DateTimeZone $timezone = null) {}
+
+    /**
+     * @psalm-mutation-free
+     *
+     * @param string $format
+     *
+     * @return (\PHP_MAJOR_VERSION is int<0, 7> ? string|false : string)
+     */
+    public function format($format) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return DateTimeZone
+     */
+    public function getTimezone() {}
+
+    /**
+     * @psalm-mutation-free
+     * @return int
+     */
+    public function getOffset() {}
+
+    /**
+     * @psalm-mutation-free
+     * @return int
+     */
+    public function getTimestamp() {}
+
+    /**
+     * @psalm-mutation-free
+     * @param bool $absolute
+     * @return DateInterval
+     */
+    public function diff(DateTimeInterface $targetObject, $absolute = false) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static|false
+     */
+    public function modify(string $modifier) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public function add(DateInterval $interval) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static|false this method can fail in case an {@see DateInterval} with relative
+     *                      week days is passed in.
+     *
+     * @see https://github.com/php/php-src/blob/534127d3b22b193ffb9511c4447584f0d2bd4e24/ext/date/php_date.c#L3157-L3160
+     */
+    public function sub(DateInterval $interval) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public function setTimezone(DateTimeZone $timezone) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public function setTime(int $hour, int $minute, int $second = 0, int $microsecond = 0) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public function setDate(int $year, int $month, int $day) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public function setISODate(int $year, int $week, int $dayOfWeek = 1) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public function setTimestamp(int $unixtimestamp) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return static
+     */
+    public static function createFromMutable(DateTime $object) {}
+
+    /**
+     * @psalm-mutation-free
+     * @return self
+     */
+    public static function createFromInterface(DateTimeInterface $object) {}
```

Comment:
This method is new, introduced in PHP 8.0.

Unsure how to declare this conditionally :D

Reply:
I'll remove it from the patch: not so interested in declaring purity of this one anyway, since it may potentially cause mutation while reading a mutable `DateTimeInterface` (unlikely, but :shrug:).

That should bring the patch to completion.

Reply:
13828771c7c544aa61a3db1ef49b7aae9e0996fd